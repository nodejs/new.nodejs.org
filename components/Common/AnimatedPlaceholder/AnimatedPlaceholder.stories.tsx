import AnimatedPlaceholder from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof AnimatedPlaceholder>;
type Meta = MetaObj<typeof AnimatedPlaceholder>;

export const Default: Story = {};

export default { component: AnimatedPlaceholder } as Meta;
