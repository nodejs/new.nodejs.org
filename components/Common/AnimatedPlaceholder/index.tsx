import styles from './index.module.scss';
import type { FC, ReactNode } from 'react';

type AnimatedPlaceholderProps = {
  children?: ReactNode;
  width?: number;
  height?: number;
};

const AnimatedPlaceholder: FC<AnimatedPlaceholderProps> = ({
  children,
  width,
  height,
}) => (
  <div
    className={styles.placeholder}
    style={{ width: `${width}px`, height: `${height}px` }}
  >
    {children || (
      <>
        <div className={styles.placeholderImage} />
        <div className={styles.placeholderText}>
          <div className={styles.placeholderTextLine} />
          <div className={styles.placeholderTextLine} />
        </div>
      </>
    )}
  </div>
);

export default AnimatedPlaceholder;
