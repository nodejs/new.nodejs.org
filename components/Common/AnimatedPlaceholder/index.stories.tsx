import AnimatedPlaceholder from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof AnimatedPlaceholder>;
type Meta = MetaObj<typeof AnimatedPlaceholder>;

export const Default: Story = {};

export const WithLoaderSkeleton: Story = {
  args: {
    children: <div className="animated-placeholder__image" />,
  },
};

export default { component: AnimatedPlaceholder } as Meta;
