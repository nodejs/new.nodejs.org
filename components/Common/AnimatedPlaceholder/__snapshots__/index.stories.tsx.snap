// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common/AnimatedPlaceholder Default smoke-test 1`] = `
<div class="AnimatedPlaceholder_placeholder__2tJoS">
  <div class="AnimatedPlaceholder_placeholderImage__pCSdr">
  </div>
  <div class="AnimatedPlaceholder_placeholderText___AwWu">
    <div class="AnimatedPlaceholder_placeholderTextLine__DG6zw">
    </div>
    <div class="AnimatedPlaceholder_placeholderTextLine__DG6zw">
    </div>
  </div>
</div>
`;

exports[`Common/AnimatedPlaceholder WithLoaderSkeleton smoke-test 1`] = `
<div class="AnimatedPlaceholder_placeholder__2tJoS">
  <div class="animated-placeholder__image">
  </div>
</div>
`;
